### PR TITLE
feat: add elasticsearch component

### PR DESCRIPTION
**Pull request by zidanehakim**
Because

- We need to integrate Elasticsearch into our data pipeline.
- This will allow us to cover data movement user stories

This commit

- Allow client to connect to their Elasticsearch database through VDP
- Auth using passwordless API Key and cloud ID through elasticsearch go sdk
- Create execution function to handle these following tasks

- [x] TASK_INDEX (INSERT)
- [x] TASK_UPDATE
- [x] TASK_SEARCH
- [x] TASK_DELETE
- [x] TASK_CREATE_INDEX
- [x] TASK_DELETE_INDEX

### CODE REVIEW

**Comment by chuang8511** [comment]
memo
https://github.com/ollama/ollama/issues/849 related issue.

File: data/elasticsearch/v0/tasks.go [hunk 1]
```diff
@@ -0,0 +1,823 @@
...
+	data := inputStruct.Data
+
+	dataJSON, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	esClient := ESIndex(*es)
+
+	_, err = esClient(indexName, bytes.NewReader(dataJSON), func(r *esapi.IndexRequest) {
```
Comment (donch1989) [inline data/elasticsearch/v0/tasks.go:207]:
```suggestion
	resp, err := esClient(indexName, bytes.NewReader(dataJSON), func(r *esapi.IndexRequest) {
```
We need to check the error in the `resp` as well. The `err` can not catch all errors.

File: data/elasticsearch/v0/config/tasks.json [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+            "template"
+          ],
+          "title": "Mappings",
+          "type": "object",
+          "required": []
+        }
+      },
+      "required": [
+        "index-name",
+        "mappings"
```
Comment (donch1989) [inline data/elasticsearch/v0/config/tasks.json:882]: Mapping should be optional.

File: data/elasticsearch/v0/config/tasks.json [hunk 3]
```diff
@@ -0,0 +1,951 @@
...
+        }
+      },
+      "required": [
+        "index-name",
+        "update-data"
+      ],
+      "instillEditOnNodeFields": [
+        "index-name",
+        "filter-sql",
+        "update"
```
Comment (donch1989) [inline data/elasticsearch/v0/config/tasks.json:265]:
```suggestion
        "update-data"
```

**Review by donch1989 — CHANGES_REQUESTED** [review]

**Pull request merged**